solutions/platinum/usaco-1332.mdx

```diff
@@ -0,0 +1,264 @@
+---
+id: usaco-1332
+source: USACO Platinum 2023 March
+title: Pareidolia
+author: Aakash Gokhale
+---
+
+## Explanation
+
+Let's first try to solve the problem without updates on a single string.
+Notice that it is always optimal to take the earliest $b$ and then the earliest $e$ and then the earliest $s$ after that and so on because it maximizes the number of places you can search to find the next letter of "bessie".
+
+Now let's try to solve the problem without updates for all substrings.
+Let $dp[i]$ be the number of positions of the string we have processed so far such that if we apply the greedy strategy described above, the next character we need to add is the $i$th character ($0$ indexed) of the string "bessie". We also need to add the substring starting from the current index. If the current letter of the string is $b$, then the next letter we need to add is $e$ so in this case we add $1$ to $dp[1]$. In every other case we need to add $1$ to $dp[0]$ because the next character we add to fill "bessie" is $b$.
```

Comment:
```suggestion
Let $dp[i]$ be the number of positions of the string we have processed so far such that if we apply the greedy strategy described above, the next character we need to add is the $i$th character ($0$ indexed) of the string "bessie". We also need to add the substring starting from the current index. If the current letter of the string is $b$, then the next letter we need to add is $e$ so in this case we add $1$ to $dp[1]$. In every other case, we need to add $1$ to $dp[0]$ because the next character we add to fill "bessie" is $b$.
```